File: AtCoder/ACP/BootCamp4b/Hard100_2/python/D_Integer.Cards.py
N, M = map(int, input().split())
A = sorted(list(map(int, input().split())))
V = sorted([list(map(int, input().split())) for _ in range(M)], key=lambda x: x[1])

idx = len(V) - 1
res = 0
for i in range(N):
    if idx < 0:
        res += A[i]
    else:
        res += max(A[i], V[idx][1])
        V[idx][0] -= 1
        if V[idx][0] == 0:
            idx -= 1
print(res)

File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Integer_Cards.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
#define MAX_M 100'000
using namespace std;
typedef long long ll;
typedef pair<ll, ll> pll;

ll N, M;
ll A[MAX_N];
ll B[MAX_M], C[MAX_M];

void solve() {
  sort(A, A + N);
  vector<pll> V;
  for (int i = 0; i < M; ++i) {
    V.push_back({C[i], B[i]});
  }
  sort(V.begin(), V.end());
  int idx = V.size() - 1;
  ll res = 0;
  for (int i = 0; i < N; ++i) {
    if (idx < 0) {
      res += A[i];
    } else {
      res += max(A[i], V[idx].first);
      --V[idx].second;
      if (V[idx].second == 0) {
        --idx;
      }
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N >> M;
  for (int i = 0; i < N; ++i) {
    cin >> A[i];
  }
  for (int i = 0; i < M; ++i) {
    cin >> B[i] >> C[i];
  }
  solve();
}